src/main/scala/viper/gobra/frontend/Parser.scala

```diff
@@ -768,6 +769,25 @@ object Parser {
         ghostPrimaryExp |
         operand
 
+    // TODO: change delimiter to { and implement required ambiguity resolution, current: !< X, ..., Z !>
+    // declaredPred!<d1, ..., dn!>
+    lazy val fpredConstruct: Parser[PPredConstructor] =
+      (idnUse ~ predConstructArgs) ^^ { p => PPredConstructor(PFPredBase(p._1), p._2) }
```

Comment:
```suggestion
      (idnUse ~ predConstructArgs) ^^ { 
        case identifier ~ args => PPredConstructor(PFPredBase(identifier).at(identifier), args) 
      }
```
I hope this compiles ^^
There are two improvements: (1) getting rid of the tuple and (2) add position information to the `PFPredBase` which is important if you create a nested node yourself (check out other occurrences of `.at(` in the parser)